csrc/index_compute.cpp

```diff
@@ -1101,26 +1101,67 @@ class UpdateLeafIndices : public IterVisitor {
     auto outer_id = merge->outer();
     auto inner_id = merge->inner();
 
+    // Nothing need to be done when mappings for the output axes
+    // already exist.
+    if (index_map_.find(out_id) != index_map_.end()) {
+      return;
+    }
+
+    if (outer_id->isBroadcast()) {
+      if (!index_map_.count(inner_id)) {
+        // Reduction axes on producer side could be visited on forward
+        //  propagation pass and current implementation does not yet
+        //  support reduciton on swizzled iterdomains, so un-indexed
+        //  reduction iterdomains are just ignored for now.
```

Comment:
Perhaps we should also mention in comments that the same assumptions are made for `bcast` iterdomains, so it reflects the check in code?
```suggestion
        //  reduction iterdomains are just ignored for now. The same applies to BroadcastOp.
```